openlr/openlr_decoder.cpp

```diff
@@ -191,4 +310,107 @@ void OpenLRDecoder::Decode(vector<LinearSegment> const & segments, uint32_t cons
   LOG(LINFO, ("Ambiguous routes:", allStats.m_moreThanOneCandidate));
   LOG(LINFO, ("Path is not reconstructed:", allStats.m_zeroCanditates));
 }
+
+void OpenLRDecoder::DecodeV2(vector<LinearSegment> const & segments,
+                             uint32_t const /* numThreads */, vector<DecodedPath> & paths)
+{
+  Graph graph(m_indexes.back(), make_unique<CarModelFactory>(m_countryParentNameGetterFn));
```

Comment:
ASSERT that m_indexes are non-empty.

Reply:
```c++
auto & index = m_indexes.back();
```
and please, use it later, instead of m_indexes.back().

Reply:
Both done.